tensorflow/python/kernel_tests/map_fn_test.py

```diff
@@ -186,6 +187,25 @@ def testMap_MultiInputSameStructureOutput(self):
     self.assertAllEqual(-nums, received[1])
     self.assertAllEqual(nums, received[2])
 
+  #@test_util.run_in_graph_and_eager_modes
+  @test_util.run_deprecated_v1
+  def testMap_autograph_indirect(self):
+    def test_function(x):
+      cond = constant_op.constant(-1)
+      if cond == 0:
+        result = x
+      else:
+        result = x
+      return result
+
+    @api.convert(recursive=False) 
```

Comment:
Use `def_function.function` here (imported as `from tensorflow.python.eager import def_function`)

Reply:
Ok with this decoration now it is failing correctly